rabbit-escape-engine/src/rabbitescape/engine/BehaviourTools.java

```diff
@@ -224,7 +223,14 @@ public boolean isFlat( Block block )
 
     public static boolean s_isFlat( Block block )
     {
-        return ( block != null && block.type == solid_flat );
+        return
+            (
+                block != null && 
+                ( 
+                   block.type == solid_flat 
+                || block.type == metal_flat 
+            ) 
+        );
```

Comment:
Not something to change as part of this pull request, but I think blocks should have a shape, and these methods should be renamed to be about the shape of blocks/interaction between shape of blocks. At the moment creating a metal ramp requires adding metal_up_ramp to a whole load of methods, whereas it should be as simple as setting a block's shape property to ramp, and its material property to metal.